csharp/lib/AsyncClient.cs

```diff
@@ -22,18 +23,35 @@ public AsyncClient(string host, UInt32 port, bool useTLS)
         }
     }
 
-    public async Task SetAsync(string key, string value)
+    private async Task<string?> command(IntPtr[] args, int argsCount, RequestType requestType)
     {
-        var message = messageContainer.GetMessageForCall(key, value);
-        SetFfi(clientPointer, (ulong)message.Index, message.KeyPtr, message.ValuePtr);
-        await message;
+        // We need to pin the array in place, in order to ensure that the GC doesn't move it while the operation is running.
+        GCHandle pinnedArray = GCHandle.Alloc(args, GCHandleType.Pinned);
+        IntPtr pointer = pinnedArray.AddrOfPinnedObject();
```

Comment:
Won't `stackalloc` be faster?

Reply:
Good readings though: [one](https://medium.com/@epeshk/the-big-performance-difference-between-arraypools-in-net-b25c9fc5e31d) [two](https://github.com/dotnet/corert/blob/master/src/System.Private.CoreLib/shared/System/Buffers/TlsOverPerCoreLockedStacksArrayPool.cs)

Reply:
dunno, you're welcome to profile :)

Reply:
@Yury-Fridlyand can you open an issue to check if later on when we'll continue C# develop?